diff --git a/node_modules/pptxgenjs/dist/pptxgen.cjs.js b/node_modules/pptxgenjs/dist/pptxgen.cjs.js
index 143dfa3..c2190ed 100644
--- a/node_modules/pptxgenjs/dist/pptxgen.cjs.js
+++ b/node_modules/pptxgenjs/dist/pptxgen.cjs.js
@@ -1595,8 +1595,10 @@ function createSlideMaster(props, target) {
                 addShapeDefinition(tgt, SHAPE_TYPE.LINE, object[key]);
             else if (MASTER_OBJECTS[key] && key === 'rect')
                 addShapeDefinition(tgt, SHAPE_TYPE.RECTANGLE, object[key]);
-            else if (MASTER_OBJECTS[key] && key === 'text')
-                addTextDefinition(tgt, [{ text: object[key].text }], object[key].options, false);
+            else if (MASTER_OBJECTS[key] && key === 'text') {
+                const _txtContent = Array.isArray(object[key].text) ? object[key].text : [{ text: object[key].text }];
+                addTextDefinition(tgt, _txtContent, object[key].options, false);
+            }
             else if (MASTER_OBJECTS[key] && key === 'placeholder') {
                 // TODO: 20180820: Check for existing `name`?
                 object[key].options.placeholder = object[key].options.name;
@@ -1604,7 +1606,8 @@ function createSlideMaster(props, target) {
                 object[key].options._placeholderType = object[key].options.type;
                 delete object[key].options.type; // remap name for earier handling internally
                 object[key].options._placeholderIdx = 100 + idx;
-                addTextDefinition(tgt, [{ text: object[key].text }], object[key].options, true);
+                const _phTxtContent = Array.isArray(object[key].text) ? object[key].text : [{ text: object[key].text }];
+                addTextDefinition(tgt, _phTxtContent, object[key].options, true);
                 // TODO: ISSUE#599 - only text is suported now (add more below)
                 // else if (object[key].image) addImageDefinition(tgt, object[key].image)
                 /* 20200120: So... image placeholders go into the "slideLayoutN.xml" file and addImage doesnt do this yet...
@@ -6554,7 +6557,7 @@ function makeXmlMaster(slide, layouts) {
     strXml +=
         '<p:clrMap bg1="lt1" tx1="dk1" bg2="lt2" tx2="dk2" accent1="accent1" accent2="accent2" accent3="accent3" accent4="accent4" accent5="accent5" accent6="accent6" hlink="hlink" folHlink="folHlink"/>';
     strXml += '<p:sldLayoutIdLst>' + layoutDefs.join('') + '</p:sldLayoutIdLst>';
-    strXml += '<p:hf sldNum="0" hdr="0" ftr="0" dt="0"/>';
+    strXml += '<p:hf sldNum="1" hdr="0" ftr="0" dt="0"/>';
     strXml +=
         '<p:txStyles>' +
             ' <p:titleStyle>' +
diff --git a/node_modules/pptxgenjs/dist/pptxgen.es.js b/node_modules/pptxgenjs/dist/pptxgen.es.js
index 449038c..d5eedac 100644
--- a/node_modules/pptxgenjs/dist/pptxgen.es.js
+++ b/node_modules/pptxgenjs/dist/pptxgen.es.js
@@ -1593,8 +1593,10 @@ function createSlideMaster(props, target) {
                 addShapeDefinition(tgt, SHAPE_TYPE.LINE, object[key]);
             else if (MASTER_OBJECTS[key] && key === 'rect')
                 addShapeDefinition(tgt, SHAPE_TYPE.RECTANGLE, object[key]);
-            else if (MASTER_OBJECTS[key] && key === 'text')
-                addTextDefinition(tgt, [{ text: object[key].text }], object[key].options, false);
+            else if (MASTER_OBJECTS[key] && key === 'text') {
+                const _txtContent = Array.isArray(object[key].text) ? object[key].text : [{ text: object[key].text }];
+                addTextDefinition(tgt, _txtContent, object[key].options, false);
+            }
             else if (MASTER_OBJECTS[key] && key === 'placeholder') {
                 // TODO: 20180820: Check for existing `name`?
                 object[key].options.placeholder = object[key].options.name;
@@ -1602,7 +1604,8 @@ function createSlideMaster(props, target) {
                 object[key].options._placeholderType = object[key].options.type;
                 delete object[key].options.type; // remap name for earier handling internally
                 object[key].options._placeholderIdx = 100 + idx;
-                addTextDefinition(tgt, [{ text: object[key].text }], object[key].options, true);
+                const _phTxtContent = Array.isArray(object[key].text) ? object[key].text : [{ text: object[key].text }];
+                addTextDefinition(tgt, _phTxtContent, object[key].options, true);
                 // TODO: ISSUE#599 - only text is suported now (add more below)
                 // else if (object[key].image) addImageDefinition(tgt, object[key].image)
                 /* 20200120: So... image placeholders go into the "slideLayoutN.xml" file and addImage doesnt do this yet...
@@ -6552,7 +6555,7 @@ function makeXmlMaster(slide, layouts) {
     strXml +=
         '<p:clrMap bg1="lt1" tx1="dk1" bg2="lt2" tx2="dk2" accent1="accent1" accent2="accent2" accent3="accent3" accent4="accent4" accent5="accent5" accent6="accent6" hlink="hlink" folHlink="folHlink"/>';
     strXml += '<p:sldLayoutIdLst>' + layoutDefs.join('') + '</p:sldLayoutIdLst>';
-    strXml += '<p:hf sldNum="0" hdr="0" ftr="0" dt="0"/>';
+    strXml += '<p:hf sldNum="1" hdr="0" ftr="0" dt="0"/>';
     strXml +=
         '<p:txStyles>' +
             ' <p:titleStyle>' +
